aptos-move/block-executor/src/types.rs

```diff
@@ -52,6 +52,23 @@ impl<T: Transaction> ReadWriteSummary<T> {
             writes: self.writes.into_iter().map(collapse).collect(),
         }
     }
+
+    pub fn keys_written(&self) -> impl Iterator<Item = &T::Key> {
+        Self::keys_except_delayed_fields(self.writes.iter())
+    }
+
+    pub fn keys_read(&self) -> impl Iterator<Item = &T::Key> {
+        Self::keys_except_delayed_fields(self.writes.iter())
+    }
+
+    fn keys_except_delayed_fields<'a>(
+        keys: impl Iterator<Item = &'a InputOutputKey<T::Key, T::Tag>>,
+    ) -> impl Iterator<Item = &'a T::Key> {
+        keys.filter_map(|k| match k {
+            InputOutputKey::Resource(key) | InputOutputKey::Group(key, _) => Some(key),
+            InputOutputKey::DelayedField(_) => None,
```

Comment:
why do we need to exclude delayedfields?

Reply:
there's no key carried there. 
the next step is to make sure the first cold read/write of/to a cold slot that contains a delayed field is a full read / write instead of a InPlaceDelayedFieldChange. I'm gonna deal with that case separately .